### ISSUE TITLE
Feature request: probe not only *.asc but also *.pubkey when installing a phar from URL

### ISSUE THREAD

**Issue by kalessil** [phar-io/phive#157]
First, huge thanks for the tool - it helps a lot to reduce my composer files and dependencies conflicts we have from time to time.

So, my request is about e.g. this case: https://github.com/infection/infection/releases/tag/0.9.0 a phar file has a public key. It would be great to support this distribution type as well.

EDIT: I could misunderstand the *.pubkey intention in that specific case, if so just close the ticket.

**Comment by theseer** [comment]
First: Thanks for the nice feedback.

Regarding the public key:
The .pubkey file is an openssl public key and not a signature file that `phive` could verify. And it's actually required for the phar to be executed. For it to work, you have to put the .pubkey next to the phar.

So in case we would allow insecure (read: non gpg signed distribution, see #134) we'd have to make phive download the .pubkey along and also place it next to the phar. This might actually be an interesting feature.

On the other hand: After all these years the openssl signature stuff exists in `ext\phar` I'm still not sure why anyone would want to use this mechanism. Imho, it's completely pointless: The only thing this method verifies is that the phar has not been modified since the signature had been made with an unknown private key accompanying the given public key. Since chances are I can simply replace the pubkey file if I have permissions to modify the phar, I fail to see where the (security) benefit in openssl signing the phar lies. Particularly since the neither the used private key nor the public key contain any means of identifying the signer.

For now, I'll close this.

### tl;dr
Yes, you misunderstood the use of *.pubkey but you raised a potential issue, even though for completely unrelated reasons.
